core/src/main/java/com/valtech/aem/saas/core/indexing/PageIndexUpdateHandler.java

```diff
@@ -0,0 +1,186 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.caconfig.ConfigurationResolver;
+import org.apache.sling.event.jobs.Job;
+import org.apache.sling.event.jobs.JobManager;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+
+@Component(name = "Search as a Service - Page Replication Event Handler",
+    immediate = true,
+    service = {EventHandler.class},
+    configurationPolicy = ConfigurationPolicy.REQUIRE,
+    configurationPid = "com.valtech.aem.saas.core.indexing.IndexUpdateHandler",
+    property = {
+        EventConstants.EVENT_TOPIC + "=" + ReplicationAction.EVENT_TOPIC,
+        EventConstants.EVENT_TOPIC + "=" + ReplicationEvent.EVENT_TOPIC
+    })
+@Designate(ocd = Configuration.class)
+@Slf4j
+public class PageIndexUpdateHandler implements EventHandler {
+
+  public static final String SAAS_CONTENT_READER = "saas-content-reader";
+
+  @Reference
+  private JobManager jobManager;
+
+  @Reference
+  private ConfigurationResolver configurationResolver;
+
+  @Reference
+  private Externalizer externalizer;
+
+  @Reference
+  private ResourceResolverProvider resourceResolverProvider;
+
+  @Reference
+  private PathExternalizerPipeline pathExternalizerPipeline;
+
+  private Configuration configuration;
+
+  @Override
+  public void handleEvent(Event event) {
+    if (!configuration.indexUpdateHandler_enable()) {
+      return;
+    }
+    ReplicationAction action = getReplicationAction(event);
+    if (action == null) {
+      return;
+    }
+    String actionPath = action.getPath();
+    log.info("Replication action {} occurred on {}", action.getType(), actionPath);
+    resourceResolverProvider.resourceResolverConsumer(SAAS_CONTENT_READER, resourceResolver ->
+        getSaasClient(resourceResolver, actionPath).ifPresent(client -> {
+          Map<String, Object> propertiesProto = getPropertiesPrototype(action, client);
+          if (configuration.indexUpdateHandler_enableAemExternalizer()) {
+            scheduleJobForPath(externalizer.publishLink(resourceResolver, actionPath), propertiesProto);
+          }
+          if (configuration.indexUpdateHandler_enableCustomPathExternalizerPipeline()) {
```

Comment:
please add a comment that this will be done after day cq externalizer is called (if enabled)